pyaedt/modeler/stackup_3d.py

```diff
@@ -0,0 +1,2284 @@
+            Variable Object.
+        """
+        return self._dielectric_x_position
+
+    @dielectric_x_position.setter
+    def dielectric_x_position(self, expression):
+        self._dielectric_x_position.expression = expression
+
+    @property
+    def dielectric_y_position(self):
+        """Stackup Starting origin y variable.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.stackup_3d.NamedVariable`
+            Variable Object.
+        """
+        return self._dielectric_x_position
+
+    @dielectric_y_position.setter
+    def dielectric_y_position(self, expression):
+        self._dielectric_y_position.expression = expression
+
+    @property
+    def dielectric_width(self):
+        """Stackup width.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.stackup_3d.NamedVariable`
+            Variable Object.
+        """
+        return self._dielectric_width
+
+    @dielectric_width.setter
+    def dielectric_width(self, expression):
+        self._dielectric_width.expression = expression
+
+    @property
+    def dielectric_length(self):
+        """Stackup length.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.stackup_3d.NamedVariable`
+            Variable Object.
+        """
+        return self._dielectric_length
+
+    @dielectric_length.setter
+    def dielectric_length(self, expression):
+        self._dielectric_length.expression = expression
+
+    @property
+    def layer_names(self):
+        """List all layer names.
+
+        Returns
+        -------
+        list
+        """
+        return self._layer_name
+
+    @property
+    def layer_positions(self):
+        """List all layer positions.
+
+        Returns
+        -------
+        List
+        """
+        return self._layer_position
+
+    @property
+    def stackup_layers(self):
+        """Dictionary of all stackup layers.
+
+        Returns
+        -------
+        dict
+        """
+        return self._stackup
+
+    @property
+    def z_position_offset(self):
+        """Elevation.
+
+        Returns
+        -------
+
+        """
+        return self._z_position_offset
+
+    @pyaedt_function_handler()
+    def add_padstack(self, name, material="copper"):
+        """Add a new padstack definition.
+
+        Parameters
+        ----------
+        name : str
+            padstack name
+        material : str, optional
+            Padstack Material
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.stackup_3d.Padstack`
+        """
+        p = Padstack(self._app, self, name, material)
+        self._padstacks.append(p)
+        return p
+
+    @pyaedt_function_handler()
+    def add_layer(self, name, layer_type="S", material="copper", thickness=0.035, fill_material="FR4_epoxy"):
+        """Add a new layer to the stackup. Can be Signal (S), ground (G) or Dielectric (D).
+        The layer is entirely filled with  fill_material. Anything will be drawn wmaterial.
```

Comment:
```suggestion
        """Add a new layer to the stackup.
        The new layer can be a signal (S), ground (G), or dielectric (D).
        The layer is entirely filled with the specified fill material. Anything will be drawn wmaterial.
```

Reply:
I don't understand the last sentence: Anything will be drawn wmaterial. What does that mean?